pkgs/io_file/test/read_as_bytes_test.dart

```diff
@@ -0,0 +1,229 @@
+        },
+      );
+    }, p.sendPort);
+    return FifoWriter._((await p.first) as SendPort);
+  }
+
+  void write(Uint8List data) {
+    _sendChannel.send(data);
+  }
+
+  void delay(Duration d) {
+    _sendChannel.send(d);
+  }
+
+  void close() {
+    _sendChannel.send(null);
+  }
+}
+
+void main() {
+  //TODO(brianquinlan): test with a very long path.
+
+  group('readAsBytes', () {
+    late String tmp;
+
+    setUp(() => tmp = createTemp('readAsBytes'));
+
+    tearDown(() => deleteTemp(tmp));
+
+    test('non-existant file', () {
+      expect(
+        () => fileSystem.readAsBytes('doesnotexist'),
+        throwsA(
+          isA<PathNotFoundException>()
+              .having((e) => e.message, 'message', 'open failed')
+              .having(
+                (e) => e.osError?.errorCode,
+                'errorCode',
+                2, // ENOENT
+              )
+              .having((e) => e.path, 'path', 'doesnotexist'),
+        ),
+      );
+    });
+
+    test('directory', () {
+      expect(
+        () => fileSystem.readAsBytes(tmp),
+        throwsA(
+          isA<FileSystemException>()
+              .having((e) => e.message, 'message', 'read failed')
+              .having(
+                (e) => e.osError?.errorCode,
+                'errorCode',
+                21, // EISDIR
+              )
+              .having((e) => e.path, 'path', tmp),
+        ),
+      );
+    });
+
+    test('symlink', () {
+      final path1 = '$tmp/file1';
+      final path2 = '$tmp/file2';
+
+      final data = randomUint8List(20);
+      File(path1).writeAsBytesSync(data);
+      Link(path2).createSync(path1);
+
+      expect(fileSystem.readAsBytes(path2), data);
+    });
+
+    test('broken symlink', () {
+      final path1 = '$tmp/file1';
+      final path2 = '$tmp/file2';
+
+      final data = randomUint8List(20);
+      File(path1).writeAsBytesSync(data);
+      Link(path2).createSync(path1);
+      File(path1).deleteSync();
+
+      expect(
+        () => fileSystem.readAsBytes(path2),
+        throwsA(
+          isA<PathNotFoundException>()
+              .having((e) => e.message, 'message', 'open failed')
+              .having(
+                (e) => e.osError?.errorCode,
+                'errorCode',
+                2, // ENOENT
+              )
+              .having((e) => e.path, 'path', path2),
+        ),
+      );
+    });
+
+    group('fifo (does not have length)', () {
+      for (var i = 0; i <= 1024; ++i) {
+        test('Read small file: $i bytes', () async {
+          final data = randomUint8List(i);
+          final path = '$tmp/file';
+          stdlibc.mkfifo(path, 438); // 0666
+
+          (await FifoWriter.create(path))
+            ..write(data)
+            ..close();
+
+          expect(fileSystem.readAsBytes(path), data);
+        });
+      }
+
+      test('many single byte reads', () async {
+        final path = '$tmp/file';
+        stdlibc.mkfifo(path, 438); // 0666
```

Comment:
what is 438?

I presume 0666 is file permissions?

Reply:
Added a better comment.